### PR TITLE
Blowout user schema; add sessions

### PR DESCRIPTION
**Pull request by matoszz**
- Added some relevant user fields
- found out you could use "mix-ins" [ref](https://entgo.io/docs/schema-mixin/) to control fields you'd always want in your schemas for stuff like created at, updated at, etc., added the audit mixin and then updated the existing schemas with it
- noted a few "TO-DO"'s to determine what model you need to use for `Byte` as many things like a password hash / key you shouldn't store or pass as a string
- added more relevant comments to the files and also appease the linter

### CODE REVIEW

File: internal/ent/schema/user.go [hunk 1]
```diff
@@ -21,26 +22,108 @@ type User struct {
...
+		field.String("avatar_local_file").
+			Comment("The user's local avatar file").
+			MaxLen(255).
+			Optional().
+			Nillable(),
+		field.Time("avatar_updated_at").
+			Comment("The time the user's (local) avatar was last updated").
+			Optional().
+			Nillable(),
+		field.Time("silenced_at").
```
Comment (sfunkhouser) [inline internal/ent/schema/user.go:69]: what does silenced mean here?

File: internal/ent/schema/user.go [hunk 2]
```diff
@@ -21,26 +22,108 @@ type User struct {
...
 				return err
 			}),
 		field.UUID("id", uuid.UUID{}).Default(uuid.New).Unique(),
-		field.Time("created_at").
-			Default(time.Now).
-			Immutable().
-			Annotations(&entsql.Annotation{
-				Default: "CURRENT_TIMESTAMP",
-			}),
+		field.String("display_name").
```
Comment (sfunkhouser) [inline internal/ent/schema/user.go:33]: Should we also have first / last name?

File: internal/ent/schema/user.go [hunk 3]
```diff
@@ -21,26 +22,108 @@ type User struct {
...
+		field.Time("avatar_updated_at").
+			Comment("The time the user's (local) avatar was last updated").
+			Optional().
+			Nillable(),
+		field.Time("silenced_at").
+			Comment("The time the user was silenced").
+			Optional().
+			Nillable(),
+		field.Time("suspended_at").
+			Comment("The time the user was silenced").
```
Comment (golanglemonade) [inline internal/ent/schema/user.go:74]:
```suggestion
			Comment("The time the user was suspended").
```

File: internal/ent/schema/user.go (hunk 2 above)
Comment (golanglemonade) [inline internal/ent/schema/user.go:33]: do we also want a first/last name?

File: internal/ent/schema/user.go (hunk 1 above)
Comment (golanglemonade) [inline internal/ent/schema/user.go:69]: what does silenced mean here?

File: internal/ent/schema/user.go (hunk 2 above)
Comment (matoszz) [reply internal/ent/schema/user.go:33]: yea - adding

File: internal/ent/schema/user.go (hunk 1 above)
Comment (matoszz) [reply internal/ent/schema/user.go:69]: I was thinking it would be useful to have w/r/t notification channels + activity feeds

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**